src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -59,10 +72,115 @@
     if (symbols.length() == 0) {
       return Collections.emptyList();
     } else if (date == null) {
-      return newClient.getHistoricalPricesWithRange(symbols, range);
+      checkAndRetrieveForRange(service, symbols, range);
+      return retrieveFromDBForRange(service, symbols, range);
     } else {
-      return newClient.getHistoricalPricesWithDate(symbols, date);
+      checkAndRetrieveForDate(service, symbols, date);
+      return retrieveFromDBForDate(service, symbols, date);
+    }
+  }
+
+  public void checkAndRetrieveForRange(HistoricalPriceDBService service, String symbol,
+      String rangeOfDays) {
+    String r = rangeOfDays.split("(?<=\\D)(?=\\d)|(?<=\\d)(?=\\D)")[0];
+    int range = 30;
+    if (r.length() != 0) {
+      range = Integer.parseInt(r);
+    }
+    LocalDate today = LocalDate.now();
+    for (int i = range; i >= 1; i--) {
+      LocalDate date = today.minusDays(i);
+      DateTimeFormatter formatters = DateTimeFormatter.ofPattern("yyyyMMdd");
```

Comment:
you don't need a separate formatter for each loop

Reply:
This should be resolved now.